### DISCUSSION TITLE
Custom Icon in Template

### DISCUSSION THREAD

**Discussion by seraphx2** [piitaya/lovelace-mushroom#508] [Ideas]
So, of course, the Template Icon allows you to set custom icons from the mdi list.
What I am wanting to do is create a "better" Person chip card and one thing I can't do is reference this: `states_attr(person._, entity_picture)`.

It would be cool if an icon is not supplied (check if string starts with `mdi:`) then to just use the path supplied by, say, an attribute (which would also allow external URLs as icons as well).

I wouldn't mind working on that if you think it isn't outside of the "scope" of your ideas about the Template card. It feels like the Template card is made so people can be more creative, and allowing this extra customizability might help with making the Template card even more powerful.

**Comment by piitaya** [comment]
I understand you want templatable entity_picture option for template card. Is that correct?

**Reply by seraphx2** [reply]
Sure. I was thinking maybe it can just be open-ended to any URL that would be cool. But if you want to limit it to info supplied by HA and specifically an entity picture or something, that works as well. It would definitely be cool to have access to that for the person entity, at least.

**Comment by mnneely** [comment]
Hi there, like everyone else, I use your Mushroom cards everywhere, especially the template card. Any more thoughts on implementing an entity_picture option to allow custom icon pictures? I want to add custom weather icons to a card. I'm aware of the `icon: weather-<state>`  .svg option, but I'm looking for more flexibility with custom .svg or .png images. Thanks very much.
